core/src/main/java/org/elasticsearch/cockroach/CockroachTransportService.java

```diff
@@ -0,0 +1,404 @@
+            this.callerNodeId = callerNodeId;
+            this.action = action;
+            this.request = request;
+        }
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public void readFrom(StreamInput in) throws IOException {
+            super.readFrom(in);
+            callerNodeId = in.readString();
+            action = in.readString();
+            request = (Request) in.readNamedWriteable(CockroachRequest.class);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeString(callerNodeId);
+            out.writeString(action);
+            out.writeNamedWriteable(request);
+        }
+
+        public String getAction() {
+            return action;
+        }
+
+        public Request getRequest() {
+            return request;
+        }
+
+        public String getCallerNodeId() {
+            return callerNodeId;
+        }
+    }
+
+    /**
+     * Transport request handler that is used to send changes in snapshot status to master
+     */
+    class StartCockroachTaskRequestHandler implements TransportRequestHandler<StartCockroachTaskRequest> {
+        @Override
+        public void messageReceived(StartCockroachTaskRequest request, final TransportChannel channel) throws Exception {
+            assert cockroachActionCoordinator != null;
+            cockroachActionCoordinator.createCockroachTask(request.getParentTask(), request.getAction(), request.getCallerNodeId(),
+                request.getRequest(), new EmptyActionListener(channel));
+
+        }
+    }
+
+    private class EmptyActionListener implements ActionListener<Empty> {
+        private final TransportChannel channel;
+
+        public EmptyActionListener(final TransportChannel channel) {
+            this.channel = channel;
+        }
+
+        @Override
+        public void onResponse(final Empty empty) {
+            try {
+                channel.sendResponse(Empty.INSTANCE);
+            } catch (IOException e) {
+                onFailure(e);
+            }
+        }
+
+        @Override
+        public void onFailure(final Exception e) {
+            try {
+                channel.sendResponse(e);
+            } catch (IOException e1) {
+                logger.warn("failed to send response", e);
+            }
+        }
+    }
+
+    private static class UpdateCockroachTaskRequest<Response extends CockroachResponse> extends TransportRequest {
+        private String uuid;
+        @Nullable
+        private Exception failure;
+        @Nullable
+        private Response response;
+
+        private UpdateCockroachTaskRequest() {
+
+        }
+
+        public UpdateCockroachTaskRequest(String uuid, Response response, Exception failure) {
+            this.uuid = uuid;
+            this.response = response;
+            this.failure = failure;
+        }
+
+        @SuppressWarnings("unchecked")
```

Comment:
the suppress warnings could be right on the line of code doing the cast instead of the whole method